OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/services/ServiceProvider.kt

```diff
@@ -80,6 +85,19 @@ class ServiceProvider(
         }
     }
 
+    // schedule to start all startable services separate threads
+    fun scheduleStartServices() {
+        val provider = this
+        coroutineScope.launch {
+            synchronized(serviceMap) {
+                for (serviceReg in serviceMap!![IStartableService::class.java]!!) {
+                    val service = serviceReg.resolve(provider) as IStartableService
```

Comment:
This looks like a bug, shouldn't we be passing `serviceReg` here instead of `provider`?

Reply:
serviceReg.resolve(IServiceProvider) requires a service provider in order resolve serviceReg. The latest commit removes the manual resolve and call getService instead.